include/zephyr/input/input.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_INPUT_H_
```

Comment:
shouldn't this be part of `include/zephyr/drivers/input.h`? I'm fine with this, though. We don't have a tradition of splitting consumer/driver APIs.

Reply:
Yeah I was actually thinking about splitting between drivers and subsys but I don't feel super strongly about it, this is a good time to discuss it. Right now the drivers would have

`include/zephyr/drivers/input.h`
`drivers/input/Kconfig`
`drivers/input/input_gpio_keys.c`
...

but the subsys stuff would go in

`include/zephyr/input/input.h`
`subsys/input/Kconfig`
`subsys/input/input.c`
`subsys/input/input_longpress.c`
...

Do you think it would be better to just keep everything under `drivers/input` and `include/drivers/input` instead? May make things simpler.